src/data/DataStore.ts

```diff
@@ -983,6 +990,13 @@ class DataStore {
                 }
             }
 
+            if (countNaN > 0 && countNaN < frameEnd - frameStart) {
+                // Append first NaN point in every bucket.
+                // It is necessary to ensure the correct order of indices.
+                newIndices[sampledIndex++] = Math.min(firstNaNIndex, nextRawIndex);
+                nextRawIndex = Math.max(firstNaNIndex, nextRawIndex);
```

Comment:
This may lead to append two `NaN` values because `nextRawIndex` is same with firstNaNIndex when it's smaller.

A possibly better solution is 
```ts
firstNaNIndex > nextRawIndex && (newIndices[sampledIndex++] = nextRawIndex);

if (countNaN > 0 && countNaN < frameEnd - frameStart) {
  newIndices[sampledIndex++] = Math.min(firstNaNIndex, nextRawIndex);
}
// Includes the case there is no NaN value, the firstNaNIndex is -1.
firstNaNIndex <= nextRawIndex && (newIndices[sampledIndex++] = nextRawIndex);
```